core/arch/arm/mm/core_mmu.c

```diff
@@ -216,10 +217,17 @@ void core_init_mmu_map(void)
 	/* map what needs to be mapped (non-null size and non INTRAM/EXTRAM) */
 	map = in;
 	while (map->type != MEM_AREA_NOTYPE) {
-		if (map->va)
-			panic();
-
-		map->va = map->pa;	/* 1-to-1 pa = va mapping */
+		if (map->va) {
+#ifdef CFG_WITH_PAGER
+			if (map->va & SMALL_PAGE_SIZE)
```

Comment:
I guess it's `SMALL_PAGE_MASK` you intended to use. Not all regions are mapped with small pages though.